packages/monaco/src/browser/monaco-snippet-suggest-provider.ts

```diff
@@ -196,7 +197,10 @@ export class MonacoSnippetSuggestProvider implements monaco.languages.Completion
             if (JsonSerializedSnippet.is(scopeOrTemplate)) {
                 accept(name, scopeOrTemplate);
             } else {
-                this.parseSnippets(scopeOrTemplate, accept);
+                // eslint-disable-next-line @typescript-eslint/no-shadow
+                for (const [name, template] of Object.entries(scopeOrTemplate)) {
+                    accept(name, template);
+                }
```

Comment:
This change means you eliminated the recursion, which means `parseSnippet` will ever only dig 2 levels deep within snippets objects (first for-loop on entries, followed by second identical for-loop.)

I'm not familiar with snippets, are there potentially deeper snippets to find? If not then I guess this is OK.

Reply:
What I found was that both in VS Code and Theia, `JsonSerializedSnippets` is defined as either a single snippet `JsonSerializedSnippet` or and array `JsonSerializedSnippet[]`. [Here](https://github.com/eclipse-theia/theia/pull/12463/files/a6a9825dabe199a01190e7c6c74371a360084400#:~:text=%7D-,export%20interface%20JsonSerializedSnippets%20%7B,%7D,-export%20interface%20JsonSerializedSnippet)

I don't think it will go deeper than 2 levels. This part is also an almost exact replica of VS Code, so hopefully we will be forewarned if there are relevant changes.